### PR TITLE
Add a high-level API for interacting with the dolphin

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: crates/flipperzero/src/dolphin/deed.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
+use flipperzero_sys as sys;
+
+/// FlipperZero apps that can generate [`Deed`]s.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum App {
```
Comment (dcoles) [inline crates/flipperzero/src/dolphin/deed.rs:5]: Do these enums need to be `#[non_exhaustive]`?

File: crates/flipperzero/src/dolphin/deed.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    fn from_raw(raw: sys::DolphinApp) -> Self {
+        match raw {
+            sys::DolphinApp_DolphinAppSubGhz => App::SubGhz,
+            sys::DolphinApp_DolphinAppRfid => App::Rfid,
+            sys::DolphinApp_DolphinAppNfc => App::Nfc,
+            sys::DolphinApp_DolphinAppIr => App::Ir,
+            sys::DolphinApp_DolphinAppIbutton => App::Ibutton,
+            sys::DolphinApp_DolphinAppBadusb => App::Badusb,
+            sys::DolphinApp_DolphinAppPlugin => App::Plugin,
+            _ => panic!("Invalid DolphinApp value {}", raw),
```
Comment (dcoles) [inline crates/flipperzero/src/dolphin/deed.rs:25]: It looks like the `DolphinApp` and `DolphinDeed` enums may be extended in the future. Ideally we shouldn't panic in this situation.

Comment (str4d) [reply crates/flipperzero/src/dolphin/deed.rs:25]: It's unlikely that new apps or deeds would be added without an SDK version bump, but I don't know if it works be a major or minor version bump. IDK if changing enums in C is considered "breaking"; I guess they are effectively non-exhaustive...

Okay, I can do the same trick that I suggested in the filesystem PR. That should let apps automatically handle new apps and deeds that the SDK provides, while preventing them from defining arbitrary app or deed enum values (or the new ones before this crate knows about them).

**Review by dcoles — APPROVED** [review]
Thanks for the PR!

**Pull request merged**